### PR TITLE
Upgrade applications/text_classifications/multi_class to use Trainer API

### PR DESCRIPTION
**Pull request by sijunhe**
### PR types
Function optimization

### PR changes
Others

### Description
Upgrade applications/text_classifications/multi_class to use Trainer API

### CODE REVIEW

File: applications/text_classification/multi_class/README.md [hunk 1]
```diff
@@ -191,62 +191,80 @@ data.txt(待预测数据文件)，需要预测标签的文本数据。
 使用CPU/GPU训练，默认为GPU训练，使用CPU训练只需将设备参数配置改为`--device "cpu"`：
 ```shell
 python train.py \
-    --dataset_dir "data" \
+    --model_name_or_path "./checkpoints/" \
```
Comment (ZeyuChen) [inline applications/text_classification/multi_class/README.md]: 这里的路径形式统一一下，避免有的有双引号有的没有

Comment (ZeyuChen) [reply applications/text_classification/multi_class/README.md]: 统一改成无双印好是否更统一

File: applications/text_classification/multi_class/README.md [hunk 2]
```diff
@@ -191,62 +191,80 @@ data.txt(待预测数据文件)，需要预测标签的文本数据。
...
+- `max_seq_length`: 最大句子长度，超过该长度的文本将被截断，不足的以Pad补全。提示文本不会被截断。
+- `num_train_epochs`: 训练轮次，使用早停法时可以选择100
+- `learning_rate`: 预训练语言模型参数基础学习率大小，将与learning rate scheduler产生的值相乘作为当前学习率。
+- `do_train`: 是否进行训练。
+- `do_eval`: 是否进行评估。
+- `eval_steps`: 评估模型的间隔步数。
+- `device`: 使用的设备，默认为`gpu`。
+- `per_device_train_batch_size`: 每次训练每张卡上的样本数量。可根据实际GPU显存适当调小/调大此配置。
+- `per_device_eval_batch_size`: 每次评估每张卡上的样本数量。可根据实际GPU显存适当调小/调大此配置。
+
```
Comment (ZeyuChen) [inline applications/text_classification/multi_class/README.md:274]: 找不到save_strategy的参数说明，整体检查下使用的配置是否有欠缺

File: applications/text_classification/multi_class/README.md (hunk 1 above)
Comment (sijunhe) [reply applications/text_classification/multi_class/README.md]: addressed

File: applications/text_classification/multi_class/README.md (hunk 2 above)
Comment (sijunhe) [reply applications/text_classification/multi_class/README.md:274]: 因为支持了Trainer的原因，增加了大量可使用的参数. 所以这里只是说明了一些用户可能调整的主要参数。剩下的参数在文档中链接了TrainingArguments的参数文档，供用户查阅

File: applications/text_classification/multi_class/train.py [hunk 3]
```diff
@@ -12,206 +12,135 @@
...
+        correct = metric.compute(
+            paddle.to_tensor(eval_preds.predictions),
+            paddle.to_tensor(eval_preds.label_ids),
+        )
+        metric.update(correct)
+        acc = metric.accumulate()
+        return {"accuracy": acc}
+
+    # Deine the early-stopping callback.
+    callbacks = [
```
Comment (lugimzzz) [inline applications/text_classification/multi_class/train.py:109]: 文档体现一下使用early stopping策略吧，还有`early_stopping_patience`设为可配置参数比较好，建议默认`early_stopping_patience`参数设置比较大一些比如8

File: applications/text_classification/multi_class/README.md [hunk 4]
```diff
@@ -188,67 +188,84 @@ data.txt(待预测数据文件)，需要预测标签的文本数据。
...
-    --early_stop \
-    --epochs 100
+    --per_device_eval_batch_size 32 \
+    --per_device_train_batch_size 32 \
+    --do_train \
+    --do_eval \
+    --metric_for_best_model accuracy \
+    --load_best_model_at_end \
+    --evaluation_strategy epoch \
+    --save_strategy epoch
```
Comment (lugimzzz) [inline applications/text_classification/multi_class/README.md]: 新增一个可配置参数`--save_total_limit 1`,限制保存的epoch数

File: applications/text_classification/multi_class/README.md [hunk 5]
```diff
@@ -188,67 +188,84 @@ data.txt(待预测数据文件)，需要预测标签的文本数据。
...
 
 如果在CPU环境下训练，可以指定`nproc_per_node`参数进行多核训练：
 ```shell
-python -m paddle.distributed.launch --nproc_per_node 8 --backend "gloo" train.py \
-    --dataset_dir "data" \
-    --device "cpu" \
+python -m paddle.distributed.launch --nproc_per_node 8 --backend gloo train.py \
+    --model_name_or_path ernie-3.0-medium-zh \
+    --data_dir ./data/ \
+    --output_dir ./checkpoints/ \
```
Comment (lugimzzz) [inline applications/text_classification/multi_class/README.md]: 下面文档的参数路径默认是`checkpoint`，尽量保持一致

File: applications/text_classification/multi_class/README.md [hunk 6]
```diff
@@ -188,67 +188,84 @@ data.txt(待预测数据文件)，需要预测标签的文本数据。
...
+    --do_eval \
+    --metric_for_best_model accuracy \
+    --load_best_model_at_end \
+    --evaluation_strategy epoch \
+    --save_strategy epoch
 ```
 
-可支持配置的参数：
+主要的配置的参数为：
+- `model_name_or_path`: 内置模型名，或者模型参数配置目录路径。默认为`ernie-3.0-base-zh`。
```
Comment (lugimzzz) [inline applications/text_classification/multi_class/README.md:262]: 添加一下可选模型列表，不是所有模型都支持后续的模型裁剪

["ernie-1.0-large-zh-cw","ernie-3.0-xbase-zh", "ernie-3.0-base-zh", "ernie-3.0-medium-zh", "ernie-3.0-micro-zh", "ernie-3.0-mini-zh", "ernie-3.0-nano-zh", "ernie-2.0-base-en", "ernie-2.0-large-en","ernie-m-base","ernie-m-large"]

File: applications/text_classification/multi_class/train.py [hunk 7]
```diff
@@ -12,206 +12,135 @@
...
+        metric = Accuracy()
+        correct = metric.compute(
+            paddle.to_tensor(eval_preds.predictions),
+            paddle.to_tensor(eval_preds.label_ids),
+        )
+        metric.update(correct)
+        acc = metric.accumulate()
+        return {"accuracy": acc}
+
+    # Deine the early-stopping callback.
```
Comment (lugimzzz) [inline applications/text_classification/multi_class/train.py]: `# Deine -># Define`

**Comment by sijunhe** [comment]
Addressed comments from @lugimzzz

**Review by lugimzzz — APPROVED** [review]
LGTM
